crates/re_data_store/tests/dump.rs

```diff
@@ -88,34 +128,45 @@ fn data_store_dump_impl(store1: &mut DataStore, store2: &mut DataStore, store3:
 
     // Fill the first store.
     for table in &tables {
+        // insert temporal
         insert_table(store1, table);
+
+        // insert timeless
+        let mut table_timeless = table.clone();
+        table_timeless.col_timelines = Default::default();
+        insert_table_with_retries(store1, &table_timeless);
     }
     sanity_unwrap(store1);
 
     // Dump the first store into the second one.
-    for table in store1.to_data_tables(None) {
-        insert_table_with_retries(store2, &table);
+    {
+        // We use a RowSet instead of a DataTable to handle duplicate RowIds.
+        let mut row_set = RowSet::default();
+        row_set.insert_tables(store1.to_data_tables(None));
+        row_set.insert_into(store2);
+        sanity_unwrap(store2);
```

Comment:
The old code would use "retrying" to generate new row-ids, which mean the resulting DataTables weren't the same anymore, hence the new `RowSet` test struct.

The old test worked because `to_dataframe` ignored rowids